workers/unity/Packages/com.improbable.gdk.core/Systems/CustomSpatialOSSendSystem.cs

```diff
@@ -13,18 +14,22 @@ public abstract class CustomSpatialOSSendSystem<T> : ComponentSystem where T : I
 
         private SpatialOSSendSystem spatialOSSendSystem;
 
-        protected WorkerBase worker;
+        public struct WorkerData
+        {
+            public readonly int Length;
+            [ReadOnly] public SharedComponentDataArray<WorkerConfig> WorkerConfigs;
+        }
+
+        [Inject] protected WorkerData WorkerConfigData;
 
         protected override void OnCreateManager(int capacity)
         {
             base.OnCreateManager(capacity);
 
-            worker = WorkerRegistry.GetWorkerForWorld(World);
-
             spatialOSSendSystem = World.GetOrCreateManager<SpatialOSSendSystem>();
             if (!spatialOSSendSystem.TryRegisterCustomReplicationSystem(typeof(T)))
             {
-                worker.LogDispatcher.HandleLog(LogType.Error, new LogEvent(CustomReplicationSystemAlreadyExists)
+                new LoggingDispatcher().HandleLog(LogType.Error, new LogEvent(CustomReplicationSystemAlreadyExists)
```

Comment:
You could add it as another generic on the system - but that doesn't feel good necessarily.

Reply:
one idea that came up in between would be to let the systems own the dispatcher instead of the worker as this is more aligned to how logging dispatchers in other projects. However, in this case it is actually possible (after realizing that you actually have access to the injected groups from the very start) and I'll change that